.github/workflows/CI.yml

```diff
@@ -53,6 +53,72 @@ jobs:
         with:
           file: lcov.info
 
+  online-test:
+    name: "Online Tests"
+    runs-on: ubuntu-latest
+    env:
+      STACK_NAME: AWSClusterManagers-jl
+      ONLINE: "docker, batch, batch-node"
+    steps:
+      - uses: actions/checkout@v2
+      - uses: julia-actions/setup-julia@v1
+        with:
+          version: "1"
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Validate AWS CloudFormation template
+        run: aws cloudformation validate-template --template-body file://test/batch.yml
+
+      # Note: Using Bors will restrict us to only having one workflow running at a given time.
+      # Because of this we can safely use a single CFN stack and just update the stack to the
+      # state of the current branch.
+      - name: Deploy to AWS CloudFormation
+        uses: aws-actions/aws-cloudformation-github-deploy@v1
+        with:
+          name: ${{ env.STACK_NAME }}
+          template: test/batch.yml
```

Comment:
Can we not have this stack just living constantly?

Reply:
That is an alternate way we could handle it but it does make it more difficult to make changes to the stack over time. For example when adding AWS Batch node support we needed to switch to an on-demand cluster vs. spot.

This way allows for contributors to this repo to make updates to the stack without having direct access to the AWS account.

Reply:
Makes sense!